### PR TITLE
FIX: make sure addFont test removes the test font

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

This should fix the intermittent failures on CI.

The issue is that the test `test_addfont_as_path` was adding an extra font (on top of what we normally discover).  When the test are run in parallel it random if this test and `test_get_font_names` run on the same worker or not. 

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

### CODE REVIEW

File: lib/matplotlib/tests/test_font_manager.py [hunk 1]
```diff
@@ -180,7 +180,16 @@ def test_addfont_as_path():
     """Smoke test that addfont() accepts pathlib.Path."""
     font_test_file = 'mpltest.ttf'
     path = Path(__file__).parent / font_test_file
-    fontManager.addfont(path)
+    try:
+        fontManager.addfont(path)
+        added, = [font for font in fontManager.ttflist
+                  if font.fname.endswith('mpltest.ttf')]
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_font_manager.py:186]:
```suggestion
                  if font.fname.endswith(font_test_file)]
```

File: lib/matplotlib/tests/test_font_manager.py [hunk 2]
```diff
@@ -180,7 +180,16 @@ def test_addfont_as_path():
...
+        fontManager.ttflist.remove(added)
+    finally:
+        to_remove = [font for font in fontManager.ttflist
+                     if font.fname.endswith('mpltest.ttf')]
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_font_manager.py:190]:
```suggestion
                     if font.fname.endswith(font_test_file)]
```

**Review by QuLogic — APPROVED** [review]
Seems good.

**Comment by tacaswell** [comment]
I intend to self-merge this on just @QuLogic 's review when CI passes to unbreak the rest of our CI.

**Pull request merged**